scaleway/resource_compute_instance_server.go

```diff
@@ -0,0 +1,360 @@
+		Zone:     zone,
+		ServerID: ID,
+	})
+	if err != nil {
+		if is404Error(err) {
+			d.SetId("")
+			return nil
+		}
+		return err
+	}
+	switch response.Server.State {
+	case instance.ServerStateStopped:
+		d.Set("state", "stopped")
+	case instance.ServerStateStoppedInPlace:
+		d.Set("state", "standby")
+	case instance.ServerStateRunning:
+		d.Set("state", "started")
+	case instance.ServerStateLocked:
+		return fmt.Errorf("server is locked, please contact Scaleway support: https://console.scaleway.com/support/tickets")
+	default:
+		return fmt.Errorf("server is in an invalid state, someone else might be executing action at the same time")
+	}
+
+	d.Set("name", response.Server.Name)
+	d.Set("image", response.Server.Image.ID)
+	d.Set("type", response.Server.CommercialType)
+	d.Set("tags", response.Server.Tags)
+	d.Set("security_group_id", response.Server.SecurityGroup.ID)
+	d.Set("enable_ipv6", response.Server.EnableIPv6)
+
+	if response.Server.PrivateIP != nil {
+		d.Set("private_ip", *response.Server.PrivateIP)
+	}
+
+	if response.Server.PublicIP != nil {
+		d.Set("public_ip", response.Server.PublicIP.Address.String())
+		d.SetConnInfo(map[string]string{
+			"type": "ssh",
+			"host": response.Server.PublicIP.Address.String(),
+		})
+	}
+
+	if response.Server.EnableIPv6 && response.Server.IPv6 != nil {
+		d.Set("public_ipv6", response.Server.IPv6.Address.String())
+	}
+
+	if vs, ok := response.Server.Volumes["0"]; ok {
+		rootVolume := flattenRootVolume(d.Get("root_volume"))
+		rootVolume[0]["volume_id"] = vs.ID
+		rootVolume[0]["size_in_gb"] = int(vs.Size / giga)
+		d.Set("root_volume", rootVolume)
+	}
+
+	// todo: set user data
+
+	return nil
+}
+
+func resourceScalewayComputeInstanceServerUpdate(d *schema.ResourceData, m interface{}) error {
+	instanceApi, zone, ID, err := getInstanceAPIWithZoneAndID(m, d.Id())
+	if err != nil {
+		return err
+	}
+
+	updateRequest := &instance.UpdateServerRequest{
+		Zone:     zone,
+		ServerID: ID,
+	}
+
+	if d.HasChange("name") {
+		updateRequest.Name = utils.String(d.Get("name").(string))
+	}
+
+	if d.HasChange("tags") {
+		updateRequest.Tags = utils.Strings(d.Get("tags").([]string))
+	}
+
+	if d.HasChange("security_group_id") {
+		updateRequest.SecurityGroup = &instance.SecurityGroupSummary{
+			ID:   d.Get("security_group_id").(string),
+			Name: getRandomName("sg"), // this value will be ignored by the API
+		}
+	}
+
+	if d.HasChange("enable_ipv6") {
+		updateRequest.EnableIPv6 = utils.Bool(d.Get("enable_ipv6").(bool))
+	}
+
+	_, err = instanceApi.UpdateServer(updateRequest)
+	if err != nil {
+		return err
+	}
+
+	if d.HasChange("state") {
+		previousState, nextState := d.GetChange("state")
+		for _, action := range stateToAction(previousState.(string), nextState.(string)) {
+			err = instanceApi.ServerActionAndWait(&instance.ServerActionAndWaitRequest{
+				Zone:     zone,
+				ServerID: ID,
+				Action:   action,
+				Timeout:  time.Minute * 10,
```

Comment:
Can we put the timeout in a constant somewhere? Might be necessary some day to change it everywhere. Or even put it in the config. But a constant will do for now.